packages/composite-checkout/demo/index.js

```diff
@@ -7,6 +7,7 @@ import '@automattic/calypso-polyfills';
 import React, { useState, useEffect, useMemo } from 'react';
 import styled from '@emotion/styled';
 import ReactDOM from 'react-dom';
+// eslint-disable-next-line import/no-extraneous-dependencies
```

Comment:
This is required because node module resolution algorithm doesn't allow a package to import itself.

An alternative would be to configure `import/no-extraneous-dependencies` to look for modules in `./node_modules` and `../../node_modules`. It is a bit nasty, opens the door to declaring dependencies on packages outside `./package.json`.

A better alternative (that I was planning to do in a separate PR) would be to add a new `package.json` to `composite-checkout/demo` (i.e. make it its own package) and have a dependency on `@automattic/composite-checkout`